torch/csrc/autograd/functions/utils.cpp

```diff
@@ -14,9 +14,7 @@ variable_list wrap_outputs(const variable_list& inputs, tensor_list&& outputs,
   if (flags.is_volatile) {
     for (auto& output : outputs) {
       if (output.defined()) {
-        auto pImpl = new VariableTensor(output);
-        pImpl->is_volatile = true;
-        result.emplace_back(pImpl, false);
+        result.emplace_back(new VariableTensor(output, false, true), false);
```

Comment:
I don't think this fixes the problem I mentioned. You only got rid of this temporary assignment, but this `VariableTensor` can still leak if the vector fails to resize (it has to increase its size before it even creates the VariableTensor. I think the right way to do it would be to construct a `Variable` here, so that it takes ownership of the `VariableTensor`, and only then will be moved into the vector